Lib/test/libregrtest/runtest.py

```diff
@@ -10,115 +12,102 @@
 
         if self.errors:
             le = len(self.errors)
             error_s = "error" + ("s" if le > 1 else "")
-            return f"{self.name} failed ({le} {error_s})"
+            return f"{self.test_name} failed ({le} {error_s})"
 
         if self.failures:
             lf = len(self.failures)
             failure_s = "failure" + ("s" if lf > 1 else "")
-            return f"{self.name} failed ({lf} {failure_s})"
-
-        return f"{self.name} failed"
-
-
-class UncaughtException(Failed):
-    def __str__(self) -> str:
-        return f"{self.name} failed (uncaught exception)"
-
-
-class EnvChanged(Failed):
-    def __str__(self) -> str:
-        return f"{self.name} failed (env changed)"
-
-    # Convert Passed to EnvChanged
-    @staticmethod
-    def from_passed(other):
-        return EnvChanged(other.name, other.duration_sec, other.xml_data)
-
-
-class RefLeak(Failed):
-    def __str__(self) -> str:
-        return f"{self.name} failed (reference leak)"
-
-
-class Skipped(TestResult):
-    def __str__(self) -> str:
-        return f"{self.name} skipped"
-
-
-class ResourceDenied(Skipped):
-    def __str__(self) -> str:
-        return f"{self.name} skipped (resource denied)"
-
-
-class Interrupted(TestResult):
-    def __str__(self) -> str:
-        return f"{self.name} interrupted"
-
-
-class ChildError(Failed):
-    def __str__(self) -> str:
-        return f"{self.name} crashed"
+            return f"{self.test_name} failed ({lf} {failure_s})"
 
+        return f"{self.test_name} failed"
 
-class DidNotRun(TestResult):
     def __str__(self) -> str:
-        return f"{self.name} ran no tests"
-
-
-class Timeout(Failed):
-    def __str__(self) -> str:
-        return f"{self.name} timed out ({format_duration(self.duration_sec)})"
+        state = self.state
+        if state == State.PASSED:
+            return f"{self.test_name} passed"
+        if state == State.FAILED:
+            return self._format_failed()
+        elif state == State.SKIPPED:
+            return f"{self.test_name} skipped"
+        elif state == State.UNCAUGHT_EXC:
+            return f"{self.test_name} failed (uncaught exception)"
+        elif state == State.REFLEAK:
+            return f"{self.test_name} failed (reference leak)"
+        elif state == State.ENV_CHANGED:
+            return f"{self.test_name} failed (env changed)"
+        elif state == State.RESOURCE_DENIED:
+            return f"{self.test_name} skipped (resource denied)"
+        elif state == State.INTERRUPTED:
+            return f"{self.test_name} interrupted"
+        elif state == State.CHILD_ERROR:
+            return f"{self.test_name} crashed"
+        elif state == State.DID_NOT_RUN:
+            return f"{self.test_name} ran no tests"
+        elif state == State.TIMEOUT:
+            return f"{self.test_name} timed out ({format_duration(self.duration)})"
+        else:
+            raise ValueError("unknown result state: {state!r}")
```

Comment:
Consider using pattern-matching here -- I find it much more elegant for this kind of usage:

```suggestion
        match self.state:
            case State.PASSED:
                return f"{self.test_name} passed"
            case State.FAILED:
                return self._format_failed()
            case State.SKIPPED:
                return f"{self.test_name} skipped"
            case State.UNCAUGHT_EXC:
                return f"{self.test_name} failed (uncaught exception)"
            case State.REFLEAK:
                return f"{self.test_name} failed (reference leak)"
            case State.ENV_CHANGED:
                return f"{self.test_name} failed (env changed)"
            case State.RESOURCE_DENIED:
                return f"{self.test_name} skipped (resource denied)"
            case State.INTERRUPTED:
                return f"{self.test_name} interrupted"
            case State.CHILD_ERROR:
                return f"{self.test_name} crashed"
            case State.DID_NOT_RUN:
                return f"{self.test_name} ran no tests"
            case State.TIMEOUT:
                return f"{self.test_name} timed out ({format_duration(self.duration)})"
            case _:
                raise ValueError("unknown result state: {state!r}")
```